AsyncDisplayKit/ASButtonNode.mm

```diff
@@ -41,33 +47,50 @@ - (instancetype)init
 
     _titleNode = [[ASTextNode alloc] init];
     _imageNode = [[ASImageNode alloc] init];
+    _backgroundImageNode = [[ASImageNode alloc] init];
+    [_backgroundImageNode setContentMode:UIViewContentModeScaleToFill];
     
+    [_titleNode setLayerBacked:YES];
+    [_imageNode setLayerBacked:YES];
+    [_backgroundImageNode setLayerBacked:YES];
+      
     _contentHorizontalAlignment = ASAlignmentMiddle;
     _contentVerticalAlignment = ASAlignmentCenter;
     
+    [self addSubnode:_backgroundImageNode];
     [self addSubnode:_titleNode];
     [self addSubnode:_imageNode];
   }
   return self;
 }
 
+- (void)setLayerBacked:(BOOL)layerBacked
+{
+  ASDisplayNodeAssert(!layerBacked, @"ASButtonNode must not be layer backed!");
```

Comment:
Interesting point with this one.  I'd actually suggest removing this — the reason is that it's not unreasonable to imagine a button that doesn't need a tap state and has a higher level node do a hitTest: to check if it falls within the button.  Buttons are often at the leaves of the hierarchy and so if all of them are forced to be view backed, it would potentially limit more optimized apps from layer-backing big chunks of the hierarchy.

The sad thing is that you're almost certainly right that as many or more developers will hit this unexpectedly and wonder why their button doesn't work, as do this intentionally.  That's mostly a documentation problem, though, and probably best addressed some other way.